packages/date/src/DatepickerInput.tsx

```diff
@@ -30,11 +30,13 @@ export const DatepickerInput = (props: DatePickerProps) => {
   }, []);
 
   // The DatepickerInput should be time and timezone agnostic,
-  // thats why we use moment().format() instead of moment().toDate().
+  // thats why we don't use moment().toDate() to get Date object.
   // moment().toDate() takes into account time and timezone and converts it
   // based on your system timezone which can result in the date change.
-  // e.g. for user who has a timezone +02:00, moment('2022-09-16T00:00+4:00').toDate() will return '2022-09-15'
-  const selectedDate = props.value && new Date(props.value?.format('YYYY-MM-DD'));
+  // e.g. if user has a timezone +02:00, moment('2022-09-16T00:00+04:00').toDate()
+  // will return September 15 instead of September 16
+  const dateObj = props.value?.toObject();
+  const selectedDate = dateObj ? new Date(dateObj.years, dateObj.months, dateObj.date) : undefined;
```

Comment:
What does `dateObj` look like? Should this use `dateObj.months - 1` since month is zero-based?

Reply:
in moment it's also zero based:
<img width="715" alt="image" src="https://user-images.githubusercontent.com/10744462/191489569-0f3b6754-6a2c-43d2-8962-4ec11f43d656.png">